plugin/src/main/java/io/jenkins/plugins/casc/SecretSourceResolver.java

```diff
@@ -19,9 +22,30 @@
     private static final String enclosedBy = "${";
     private static final String enclosedIn = "}";
     private static final char escapedWith = '^';
-    private static final String defaultDelimiter = ":-";
 
     private static final Logger LOGGER = Logger.getLogger(SecretSourceResolver.class.getName());
+    private static final StringSubstitutor NULL_SUBSTITUTOR = new StringSubstitutor(
+        UnresolvedLookup.INSTANCE);
+    private static final StringSubstitutor SUBSTITUTOR = new StringSubstitutor(
+        StringLookupFactory.INSTANCE.interpolatorStringLookup(
+            ImmutableMap.of(
+                "base64", StringLookupFactory.INSTANCE.base64EncoderStringLookup(),
+                "file", FileStringLookup.INSTANCE
```

Comment:
One could argue the decoding of base64 could be useful as well 😅
Though to reduce confusion or potential addition, we should establish a good default.
```suggestion
                "base64", StringLookupFactory.INSTANCE.base64EncoderStringLookup(),
                "string", StringLookupFactory.INSTANCE.base64DecoderStringLookup(),
                "file", FileStringLookup.INSTANCE
```

or more literal?

```suggestion
                "base64encode", StringLookupFactory.INSTANCE.base64EncoderStringLookup(),
                "base64decode", StringLookupFactory.INSTANCE.base64DecoderStringLookup(),
                "file", FileStringLookup.INSTANCE
```

Reply:
meh, add it if we need it